ipfshttpclient/client/__init__.py

```diff
@@ -106,6 +107,10 @@ def connect(
 	All parameters are identical to those passed to the constructor of the
 	:class:`~ipfshttpclient.Client` class.
 	"""
+
+	if headers is None:
+		headers = {}
+
```

Comment:
Quick question: What is the benefit of splitting this out over keeping it in inline in the declaration?

Reply:
Mutable default values are dangerous because if the object gets mutated, the default value is different on the next call.

https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument
https://towardsdatascience.com/python-pitfall-mutable-default-arguments-9385e8265422

Reply:
I'd like to argue that its not a problem here since it will *never* mutate the given `headers` object without copying it. That could break things like:

```py
headers = {…}
client.add("abc", headers=headers)
client.add("xyz", headers=headers)
```

What should definitely change here is the definition of `headers_t`, which currently refers to `ty.Dict[…]` but should actually be `ty.Mapping[…]`. The `{}` here just represents an arbitrary empty mapping; if [`frozenmap`](https://www.python.org/dev/peps/pep-0603/) were a thing, I'd have used that instead.

If the code every mutates `headers` it is already wrong for the above reason.